app/src/main/java/com/engineering/dokkan/MainFragment.java

```diff
@@ -0,0 +1,97 @@
+package com.engineering.dokkan;
+
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import com.engineering.dokkan.view.base.BaseFragment;
+
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class MainFragment extends BaseFragment {
+   private Button btn_home , btn_msg , btn_notification , btn_profile , btn_askQ ;
+
+
+    public MainFragment() {
+        // Required empty public constructor
+    }
+
+
+    @Override
+    public int getLayoutId() {
+        return R.layout.fragment_main;
+    }
+
+    NavController getNavController(){
+        return Navigation.findNavController(getActivity() ,R.id.nav_host_fragment);
+    }
+
+    @Override
+    public void initializeViews(View view) {
+        initialization(view);
+        btn_home.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
```

Comment:
7abiby make initialization for views inside this method and setting listeners inside setListeners method